src/taskgraph/util/taskcluster.py

```diff
@@ -75,6 +76,66 @@ def get_root_url(use_proxy):
     return liburls.normalize_root_url(PRODUCTION_TASKCLUSTER_ROOT_URL)
 
 
+@functools.lru_cache(maxsize=None)
+def get_taskcluster_queue():
+    if "TASKCLUSTER_PROXY_URL" in os.environ:
+        queue_options = {"rootUrl": os.environ["TASKCLUSTER_PROXY_URL"]}
+    else:
+        queue_options = taskcluster.optionsFromEnvironment()
+
+    return taskcluster.Queue(queue_options)
+
+
+@functools.lru_cache(maxsize=None)
+def get_taskcluster_index():
+    if "TASKCLUSTER_PROXY_URL" in os.environ:
+        index_options = {"rootUrl": os.environ["TASKCLUSTER_PROXY_URL"]}
+    else:
+        index_options = taskcluster.optionsFromEnvironment()
+
+    return taskcluster.Index(index_options)
+
+
+@functools.lru_cache(maxsize=None)
+def get_taskcluster_auth():
+    if "TASKCLUSTER_PROXY_URL" in os.environ:
+        auth_options = {"rootUrl": os.environ["TASKCLUSTER_PROXY_URL"]}
+    else:
+        auth_options = taskcluster.optionsFromEnvironment()
+
+    return taskcluster.Auth(auth_options)
+
+
+@functools.lru_cache(maxsize=None)
+def get_taskcluster_purge_cache():
+    if "TASKCLUSTER_PROXY_URL" in os.environ:
+        purge_cache_options = {"rootUrl": os.environ["TASKCLUSTER_PROXY_URL"]}
+    else:
+        purge_cache_options = taskcluster.optionsFromEnvironment()
+
+    return taskcluster.PurgeCache(purge_cache_options)
+
+
+@functools.lru_cache(maxsize=None)
+def get_taskcluster_notify():
+    if "TASKCLUSTER_PROXY_URL" in os.environ:
+        notify_options = {"rootUrl": os.environ["TASKCLUSTER_PROXY_URL"]}
+    else:
+        notify_options = taskcluster.optionsFromEnvironment()
+
+    return taskcluster.Notify(notify_options)
+
+
+@functools.lru_cache(maxsize=None)
+def get_taskcluster_secret():
+    if "TASKCLUSTER_PROXY_URL" in os.environ:
+        secret_options = {"rootUrl": os.environ["TASKCLUSTER_PROXY_URL"]}
+    else:
+        secret_options = taskcluster.optionsFromEnvironment()
+
+    return taskcluster.Secrets(secret_options)
```

Comment:
Might be nicer to do something like:
https://github.com/mozilla-releng/fxci-config/blob/50f95de0057cb77b87d5f0cee12560c335338a40/taskcluster/fxci_config_taskgraph/util/integration.py#L38

At the very least factoring out the options into some kind of `_get_options` helper would help cut down some duplication.